plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -418,44 +366,47 @@ func (t *Tagger) refreshLoopToUpdateTagsAndVolumes() {
 	}
 }
 
-// updateVolumes calls EC2 describe volume
-func (t *Tagger) updateVolumes() error {
-	if t.ebsVolume == nil {
-		t.ebsVolume = NewEbsVolume()
+/*
+	Retrieve metadata from IMDS and use these metadata to:
+	* Extract InstanceID, ImageID, InstanceType to create custom dimension for collected metrics
+	* Extract InstanceID to retrieve Instance's Volume and Tags
+	* Extract Region to create aws session with custom configuration
+	For more information on IMDS, please follow this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
+*/
+func (t *Tagger) deriveEC2MetadataFromIMDS() error {
+	for _, tag := range t.EC2MetadataTags {
+		switch tag {
+			case mdKeyInstanceId:
+				t.ec2MetadataLookup.instanceId = true
```

Comment:
If instance ID is "required" then what was the original purpose of this lookup check? And why are we keeping it?

Reply:
As you already know, the EC2Tagger's purpose is to append the dimensions (which can be scraped by tags and metadata from IMDS) to the metrics. Therefore, the `ec2MetadataLookup` is to ensured the customer can append the [dimensions to their needs](https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/processors/ec2tagger/ec2tagger.go#L167-L175).

**For example:** The below configuration will ignore `InstanceType` but will collect ImageID and add the dimension ImageId to the metrics, while InstanceId still be connected but only for retrieving tags and volumes.
```
"append_dimensions": {
			"ImageId": "${aws:ImageId}",
},
```
While if customer does have `"InstanceId": "${aws:InstanceId}",`, the CloudWatchAgent will also append the InstanceId and ImageId to the metrics as their dimension.